packages/expo-image/ios/ImageUtils.swift

```diff
@@ -102,23 +102,16 @@ func shouldDownscale(image: UIImage, toSize size: CGSize, scale: Double) -> Bool
 /**
  Resizes the animated image to fit in the given size and scale.
  */
-func resize(animatedImage image: UIImage, toSize size: CGSize, scale: Double) async -> UIImage {
-  // If there are no image frames, only resize the main image.
-  guard let images = image.images else {
-    return resize(image: image, toSize: size, scale: scale)
-  }
-
-  // Resize all animated image frames.
-  let resizedImages = await concurrentMap(images) { image in
-    return resize(image: image, toSize: size, scale: scale)
+func resize(animatedImage image: UIImage, toSize size: CGSize, scale: Double) -> UIImage {
+  if image.sd_isAnimated,
+    let animatedImage = image as? AnimatedImage {
+    let animatedCoder = ResizedAnimatedCoder(image: animatedImage, size: size, scale: scale)
+    if let result = AnimatedImage(animatedCoder: animatedCoder, scale: scale) {
```

Comment:
It's a clever idea to use custom image class (which will receive `animatedImageFrameAtIndex` call to get frame) with post-process transform